teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,464 @@
+| ---                     | ---                 | ---                      | ---                                                  | ---          |
+| TEP Process Stewardship | Tekton Contributors | Tekton Governing members | Tekton Contributors                                  | Community    |
+| Enhancement delivery    | Enhancement Owner   | Project(s) OWNERs        | Enhancement Implementer(s) (may overlap with Driver) | Community    |
+
+In a nutshell, this means:
+- Updates on the TEP process are driven by contributors and approved
+  by the tekton governing board.
+- Enhancement proposal are driven by contributors, and approved by the
+  related project(s) owners.
+
+## Reference-level explanation
+
+### What type of work should be tracked by a TEP
+
+The definition of what constitutes an "enhancement" is a foundational
+concern for the Tekton project. Roughly any Tekton user or operator
+facing enhancement should follow the TEP process. If an enhancement
+would be described in either written or verbal communication to anyone
+besides the TEP author or developer, then consider creating a
+TEP. This means any change that may impact any other community project
+in a way should be proposed as a TEP. Those changes can be for
+technical reason, adding or removing (deprecating then removing)
+features.
+
+Similarly, any technical effort (refactoring, major architectural
+change) that will impact a large section of the development community
+should also be communicated widely. The TEP process is suited for this
+even if it will have zero impact on the typical user or operator.
+
+Let's list a few enhancement that happened before this process (or are
+happening), that would have required a TEP:
+
+- Failure strategies using runOn 🎉 [tektoncd/pipeline#2094](https://github.com/tektoncd/pipeline/issues/2094)
+- Expose v1beta1 to the world ⛈ [tektoncd/pipeline#2035](https://github.com/tektoncd/pipeline/issues/2035)
+- Initial Implementation of conditionals [tektoncd/pipeline#1093](https://github.com/tektoncd/pipeline/issues/1093)
+- Adding new ways to resolve Task/Pipeline definition, using OCI
+  images or other means ([tektoncd/pipeline#2298](https://github.com/tektoncd/pipeline/issues/2298),
+  [tektoncd/pipeline#1839](https://github.com/tektoncd/pipeline/issues/1839))
+- Improve UX of getting credentials into Tasks ([tektoncd/pipeline#2343](https://github.com/tektoncd/pipeline/issues/2343))
+
+Let's also list some changes or features that are not yet in progress
+and could benefit from a TEP:
+
+- Pipeline Resources re-design, a.k.a. bring `PipelineResource` to
+  v1beta1
+- Bring `Conditions` to `v1beta1` or rewrite them differently
+- Automated releases across projects
+- CI setup using Tekton on Tekton (aka the /dogfooding/ project)
+- Serving new API version (v1beta2, v1, …) on `tektoncd/pipeline`
+- Beta APIs on `tektoncd/triggers`
+- Local-to-Tekton feature on `tektoncd/cli` (aka use local source to
+  execute a Pipeline in the cluster)
+
```

Comment:
what do you think about including examples of things that would NOT require a TEP, to give an idea of where the line is?

i'm guessing theses would NOT need TEPs:
- Propagate annotations from Conditions to TaskRuns/Pods: https://github.com/tektoncd/pipeline/pull/2608
- Improvments on Pipeline cancel https://github.com/tektoncd/pipeline/pull/2543

I'm not quite sure about these, I kinda hope they wouldnt need TEPs:
- Add variable substitution for PVC name: https://github.com/tektoncd/pipeline/pull/2506
- Emit events when we fail to update the taskrun https://github.com/tektoncd/pipeline/pull/2526

I _could_ imagine requiring TEPs for the last 2, but I feel like that might be more work than it's worth? Maybe if you can describe the problem and solution in the description of a GitHub issue, that's an indication a TEP is not required? Maybe TEPs are for the sort of features that we'd be asking for design docs for?

Reply:
> Maybe TEPs are for the sort of features that we'd be asking for design docs for?

In a gist, yes :wink: